### PR TITLE
Sue Creek: enable booting by loading over SPI

### PR DESCRIPTION
**Pull request by lyakh**
Tested: boots. To create an image something like this can be used:

```
dd if=/dev/zero bs=1 count=$(expr 9216 - $(stat --print=%s src/arch/xtensa/boot_ldr-sue.bin)) | \
cat src/arch/xtensa/boot_ldr-sue.bin - > /tmp/boot_ldr-sue.bin
cat /tmp/boot_ldr-sue.bin src/arch/xtensa/sof-sue.ri > /tmp/sue.ri
```

### CODE REVIEW

File: src/platform/suecreek/include/platform/memory.h [hunk 1]
```diff
@@ -331,19 +334,28 @@
...
 
 /* TODO: set this value */
-#define BOOT_LDR_MANIFEST_BASE		0x55aa55aa
+#define MAN_DESC_OFFSET			0x2000	/* from rimage/manifest.h */
+/*
+ * This is a temporary hack until rimage is updated to handle Sue Creek:
+ * currently the resuult of "xtensa-...-objdump -O binary sof-sue" is about
+ * 8250 bytes, reserve 9K for it and create a firmware image with something like
+ * dd if=/dev/zero bs=1 count=$(expr 9216 - $(stat --print=%s boot_ldr-sue.bin)) >> boot_ldr-sue.bin
+ * cat boot_ldr-sue.bin sof-sue.ri > /tmp/sue.img
```
Comment (lgirdwood) [inline src/platform/suecreek/include/platform/memory.h]: Sorry, can you also include the rimage updates here too so we dont need any hacks.

File: src/arch/xtensa/up/xtos/reset-vector.S [hunk 2]
```diff
@@ -581,7 +581,7 @@ unpackdone:
 
 #if CONFIG_BOOT_LOADER
 		/*ToDo refine the _start*/
-		movi a0, SOF_TEXT_START
+		movi a0, SOF_TEXT_BASE
```
Comment (lgirdwood) [inline src/arch/xtensa/up/xtos/reset-vector.S]: #ifdef SUECREEK ? Why is this needed.

Comment (lyakh) [reply src/arch/xtensa/up/xtos/reset-vector.S]: This is needed because that's where the linker script puts the text section:
```
  sof_text :
        org = SOF_TEXT_BASE,
        len = SOF_TEXT_SIZE,
```
```
#define L2_VECTOR_SIZE		0x2000

#define SOF_TEXT_START		(HP_SRAM_BASE + L2_VECTOR_SIZE)
#define SOF_TEXT_START_SIZE	0x400
#define SOF_TEXT_BASE		(SOF_TEXT_START + SOF_TEXT_START_SIZE)
```
Output of `objdump -h`:
` 10 .text         00012f62  be002400  be002400  00001400  2**4`
The other 3 platforms, that use BOOTLOADER:
`AM_CONDITIONAL(BUILD_BOOTLOADER,  test "$FW_NAME" = "apl" -o "$FW_NAME" = "cnl" -o "$FW_NAME" = "icl" -o "$FW_NAME" = "sue")`
have:
```
#define SOF_TEXT_BASE		(SOF_TEXT_START)
```
So, so far no `#ifdef ` seems to be needed?

File: src/platform/suecreek/include/platform/memory.h (hunk 1 above)
Comment (lyakh) [reply src/platform/suecreek/include/platform/memory.h]: That would take me some time. Should I do that now or after IPC?

**Comment by lgirdwood** [comment]
#762 adds the functionality to rimage.
